tests/PHPStan/Analyser/nsrt/non-falsy-string.php

```diff
@@ -104,8 +104,17 @@ function stringFunctions(string $s, $nonFalsey, $arrayOfNonFalsey, $nonEmptyArra
 
 		assertType('non-falsy-string', preg_quote($nonFalsey));
 
-		assertType('non-falsy-string', sprintf($nonFalsey));
-		assertType('non-falsy-string', vsprintf($nonFalsey, []));
+		assertType('string', sprintf($nonFalsey));
+		assertType("'foo'", sprintf('foo'));
+		assertType("string", sprintf(...$arr));
+		assertType("non-falsy-string", sprintf('%s', ...$arr)); // should be 'string'
```

Comment:
the wrong `non-falsy-string` types will be fixed after we integrated the remaining parts of https://github.com/phpstan/phpstan-src/pull/3168